src/OSPSuite.Core/Domain/Module.cs

```diff
@@ -39,132 +40,37 @@ public override void UpdatePropertiesFrom(IUpdatable source, ICloneManager clone
-         EventGroups = cloneManager.Clone(sourceModule.EventGroups);
-
-         sourceModule.MoleculeStartValuesCollection.Each(x => _moleculeStartValuesCollection.Add(cloneManager.Clone(x)));
-         sourceModule.ParameterStartValuesCollection.Each(x => _parameterStartValuesCollection.Add(cloneManager.Clone(x)));
-
+         _buildingBlocks.AddRange(sourceModule.BuildingBlocks.Select(cloneManager.Clone));
          ExtendedProperties.UpdateFrom(sourceModule.ExtendedProperties);
       }
 
-      public void AddParameterStartValueBlock(ParameterStartValuesBuildingBlock parameterStartValuesBuildingBlock)
-      {
-         _parameterStartValuesCollection.Add(parameterStartValuesBuildingBlock);
-      }
-
-      public void AddMoleculeStartValueBlock(MoleculeStartValuesBuildingBlock moleculeStartValuesBuildingBlock)
-      {
-         _moleculeStartValuesCollection.Add(moleculeStartValuesBuildingBlock);
-      }
-
-      public void RemoveParameterStartValueBlock(ParameterStartValuesBuildingBlock parameterStartValuesBuildingBlock)
-      {
-         _parameterStartValuesCollection.Remove(parameterStartValuesBuildingBlock);
-      }
-
-      public void RemoveMoleculeStartValueBlock(MoleculeStartValuesBuildingBlock moleculeStartValuesBuildingBlock)
-      {
-         _moleculeStartValuesCollection.Remove(moleculeStartValuesBuildingBlock);
-      }
-
-      public void AddBuildingBlock(IBuildingBlock buildingBlock)
+      public void Add(IBuildingBlock buildingBlock)
       {
-         switch (buildingBlock)
+         if (isSingle(buildingBlock))
          {
-            case MoleculeBuildingBlock molecule:
-               Molecules = molecule;
-               break;
-            case ReactionBuildingBlock reaction:
-               Reactions = reaction;
-               break;
-            case SpatialStructure spatialStructure:
-               SpatialStructure = spatialStructure;
-               break;
-            case PassiveTransportBuildingBlock passiveTransport:
-               PassiveTransports = passiveTransport;
-               break;
-            case EventGroupBuildingBlock eventGroup:
-               EventGroups = eventGroup;
-               break;
-            case ObserverBuildingBlock observer:
-               Observers = observer;
-               break;
-            case ParameterStartValuesBuildingBlock parameterStartValues:
-               AddParameterStartValueBlock(parameterStartValues);
-               break;
-            case MoleculeStartValuesBuildingBlock moleculeStartValues:
-               AddMoleculeStartValueBlock(moleculeStartValues);
-               break;
-            case null:
-               return;
-            default:
-               throw new OSPSuiteException(Error.BuildingBlockTypeNotSupported(buildingBlock.Name));
+            var type = buildingBlock.GetType();
+            var existingBuildingBlock = _buildingBlocks.FirstOrDefault(x => x.IsAnImplementationOf(type));
```

Comment:
this is for the case tha tI am adding another Reaction. This should not happen really but better be safe